Todo redux

Ishita382 opened the pull request.

Ishita382 committed:
modified:   src/App.css
	modified:   src/App.js
	new file:   src/Components/Button.js
    new file:   src/Components/Footer.js
	new file:   src/Components/Input.js
	new file:   src/Components/TodoApp.js
	new file:   src/Components/TodoItem.js

Ishita382 committed:
deleted:    src/App.test.js

Ishita382 committed:
modified:   src/index.js
	deleted:    src/logo.svg
	deleted:    src/reportWebVitals.js

Ishita382 committed:
modified:   package-lock.json
	modified:   package.json
	modified:   src/Components/Input.js
	modified:   src/Components/TodoApp.js
	modified:   src/Components/TodoItem.js
	src/actions/
	src/reducers/
	src/store.js

Ishita382 committed:
modified:   package-lock.json
	modified:   package.json
	modified:   src/Components/Input.js
	modified:   src/Components/TodoApp.js
	modified:   src/Components/TodoItem.js
	new file:   src/actions/actionTypes.js
	new file:   src/actions/index.js
	new file:   src/reducers/todoReducers.js
	new file:   src/store.js

shreyam1006 commented on src/App.css:
```diff
@@ -1,38 +1,36 @@
-    transform: rotate(360deg);
-  }
-}
+::placeholder{
```
eliminate all the css

shreyam1006 commented on src/Components/Button.js:
```diff
@@ -0,0 +1,33 @@
+    }
+
+
+export default Arrowbutton;
```
Add extra line

shreyam1006 commented on src/Components/Button.js:
```diff
@@ -0,0 +1,33 @@
+            return (
+                <Box>
+        {props.countAll ?  (
+          <CustomButton onClick={props.checkAll}>
```
destructure props and then use them in the component, also add null checks

shreyam1006 commented on src/Components/Footer.js:
```diff
@@ -0,0 +1,77 @@
+    color: #777;
+    `
+
+function Footer(props){
```
destructure props first

shreyam1006 commented on src/Components/Footer.js:
```diff
@@ -0,0 +1,77 @@
+            <CustomBox>
+            <Items>{props.left} {props.left===1?"item":"items"} left</Items>
+            <Buttons>
+            <CustomizedButton onClick={()=>props.change("all")} >All</CustomizedButton>
```
Create constants for all active completed

shreyam1006 commented on src/Components/Footer.js:
```diff
@@ -0,0 +1,77 @@
+    
+}
+
+export default Footer;
```
Line

shreyam1006 commented on src/Components/TodoApp.js:
```diff
@@ -0,0 +1,113 @@
+            ? completeArray
+            : all,
+        list: all.length,
+        completed: completeArray.length,
```
no need to keep separate counts for all, just count the number of todos in the listToMap array directly in the UI

shreyam1006 commented on src/Components/TodoApp.js:
```diff
@@ -0,0 +1,113 @@
+          <Input 
+            value={newToDo}
+            change={(e) => dispatch(newTodo(e.target.value))}
+            enter={(e) => addItem(e, newToDo)}
```
why do you need to put newTodo in the arguments if it already exists in the reducer?

shreyam1006 commented on src/reducers/todoReducers.js:
```diff
@@ -0,0 +1,117 @@
+  editToDo: "",
+};
+
+const todoReducers = (state = initialState, action) => {
```
destructure, type and payload from action

shreyam1006 commented on src/reducers/todoReducers.js:
```diff
@@ -0,0 +1,117 @@
+
+    case ALL:
+      const temp =
+        action.payload.completed < action.payload.list
```
why this?

shreyam1006 commented on src/reducers/todoReducers.js:
```diff
@@ -0,0 +1,117 @@
+      };
+
+    case EDIT:
+      const edittodo = state.all.map((item) =>
```
camel case

shreyam1006 commented on src/reducers/todoReducers.js:
```diff
@@ -0,0 +1,117 @@
+      const edittodo = state.all.map((item) =>
+        item.id === action.payload.id
+          ? { ...item, action: action.payload.editToDo, isEdit: false }
+          : item
```
no return statement in map?

shreyam1006 commented on src/reducers/todoReducers.js:
```diff
@@ -0,0 +1,117 @@
+        mode: action.payload.mode,
+      };
+
+    case CLEAR:
```
this should be clear completed

Ishita382 committed:
modified:   src/App.css
	modified:   src/Components/Button.js
	modified:   src/Components/TodoApp.js
	modified:   src/reducers/todoReducers.js

Ishita382 committed:
modified:   src/App.css
	modified:   src/Components/Button.js
	modified:   src/Components/Footer.js
	modified:   src/Components/Input.js
	modified:   src/Components/TodoApp.js
	modified:   src/Components/TodoItem.js
	modified:   src/actions/actionTypes.js
	modified:   src/actions/index.js
	modified:   src/reducers/todoReducers.js

FizaAggarwal commented on src/App.js:
```diff
@@ -1,25 +1,12 @@
-          Learn React
-        </a>
-      </header>
+    <div >
```
Use Box instead.

FizaAggarwal commented on src/App.js:
```diff
@@ -1,25 +1,12 @@
 }
 
-export default App;
+export default App;
```
Add extra line at the end.

FizaAggarwal commented on src/Components/TodoItem.js:
```diff
@@ -0,0 +1,90 @@
+  const {toggle, todo, value, change, editInput, enter} = props;
+      return (
+        <CustomizedBox>
+        {!props.todo.isEdit && (
```
Use destructured props variable.

FizaAggarwal commented on src/reducers/todoReducers.js:
```diff
@@ -0,0 +1,117 @@
+        all: [
+          ...state.all,
+          {
+            id: uuid(),
```
store it in a variable and then use variable to assign the value.

FizaAggarwal commented on src/store.js:
```diff
@@ -0,0 +1,10 @@
+  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
+});
+
+export default store;
```
add extra line.

Ishita382 committed:
modified:   src/Components/TodoApp.js

Ishita382 committed:
modified:   src/Components/TodoApp.js
	modified:   src/Components/TodoItem.js
	modified:   src/reducers/todoReducers.js
	modified:   src/store.js

Ishita382 committed:
modified:   src/Components/TodoItem.js
	modified:   src/reducers/todoReducers.js
	modified:   src/store.js
